llvm/lib/Transforms/Utils/SimplifyLibCalls.cpp

```diff
@@ -2603,13 +2604,27 @@ Value *LibCallSimplifier::optimizeSqrt(CallInst *CI, IRBuilderBase &B) {
   return copyFlags(*CI, FabsCall);
 }
 
-// TODO: Generalize to handle any trig function and its inverse.
-Value *LibCallSimplifier::optimizeTan(CallInst *CI, IRBuilderBase &B) {
+Value *LibCallSimplifier::optimizeTrigInverionPairs(CallInst *CI,
+                                                    IRBuilderBase &B) {
+  static const std::map<llvm::StringRef, LibFunc> trigInverses = {
```

Comment:
Don't use std::map. Can create an alphabetically sorted constant table and search with lower_bound

Reply:
`llvm::StringSwitch` is also a workable option.